src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString.split(';');
+  const clearStyles = styles
+    .map((style) => style.trim())
+    .filter((style) => style.length > 0);
+  const result = clearStyles.reduce((prev, style) => {
+    const property = style.split(':')[0].trim();
+    const value = style.split(':')[1].trim();
+
+    return { ...prev, [property]: value };
```

Comment:
Using the spread operator inside the reduce callback for each iteration can lead to performance issues, especially with a large number of styles. Each iteration creates a new object which can be inefficient. Consider using the previous object directly and assigning new properties to it.